server/src/main/java/org/apache/iotdb/db/query/udf/builtin/UDTFContinuouslySatisfy.java

```diff
@@ -0,0 +1,240 @@
+    setDefaultValue();
+    validator
+        .validateInputSeriesNumber(1)
+        .validateInputSeriesDataType(
+            0,
+            TSDataType.INT32,
+            TSDataType.INT64,
+            TSDataType.FLOAT,
+            TSDataType.DOUBLE,
+            TSDataType.BOOLEAN)
+        .validate(
+            args -> (Long) args[1] >= (Long) args[0],
+            "max can not be smaller than min.",
+            validator.getParameters().getLongOrDefault("min", defaultMin),
+            validator.getParameters().getLongOrDefault("max", defaultMax))
+        .validate(
+            min -> (Long) min >= defaultMin,
+            "min can not be smaller than " + defaultMin + ".",
+            validator.getParameters().getLongOrDefault("min", defaultMin));
+  }
+
+  protected abstract void setDefaultValue();
+
+  protected void setDefaultMax(Long defaultMax) {
+    this.defaultMax = defaultMax;
+  }
+
+  protected void setDefaultMin(Long defaultMin) {
+    this.defaultMin = defaultMin;
+  }
+
+  @Override
+  public void beforeStart(UDFParameters parameters, UDTFConfigurations configurations)
+      throws MetadataException, UDFInputSeriesDataTypeNotValidException {
+    satisfyValueCount = 0L;
+    satisfyValueStartTime = 0L;
+    satisfyValueLastTime = -1L;
+    intervals = new ArrayList<>(0);
+
+    dataType = parameters.getDataType(0);
+    min = parameters.getLongOrDefault("min", defaultMin);
+    max = parameters.getLongOrDefault("max", defaultMax);
+    configurations
+        .setAccessStrategy(new RowByRowAccessStrategy())
+        .setOutputDataType(TSDataType.INT64);
+  }
+
+  @Override
+  public void transform(Row row, PointCollector collector)
+      throws IOException, UDFInputSeriesDataTypeNotValidException {
+    switch (dataType) {
+      case INT32:
+        transformInt(row.getTime(), row.getInt(0));
+        break;
+      case INT64:
+        transformLong(row.getTime(), row.getLong(0));
+        break;
+      case FLOAT:
+        transformFloat(row.getTime(), row.getFloat(0));
+        break;
+      case DOUBLE:
+        transformDouble(row.getTime(), row.getDouble(0));
+        break;
+      case BOOLEAN:
+        transformBoolean(row.getTime(), row.getBoolean(0));
+        break;
+      default:
+        // This will not happen
+        throw new UDFInputSeriesDataTypeNotValidException(
+            0, dataType, TSDataType.INT32, TSDataType.INT64, TSDataType.FLOAT, TSDataType.DOUBLE);
+    }
+  }
+
+  protected void transformDouble(long time, double value) {
+    if (satisfyDouble(value) && satisfyValueCount == 0L) {
+      satisfyValueCount++;
+      satisfyValueStartTime = time;
+      satisfyValueLastTime = time;
+    } else if (satisfyDouble(value)) {
+      satisfyValueCount++;
+      satisfyValueLastTime = time;
+    } else {
+      if (getRecord() >= min && getRecord() <= max && satisfyValueCount > 0) {
+        intervals.add(new Pair<>(satisfyValueStartTime, getRecord()));
```

Comment:
Try to output the result to PointCollector here. It's not recommended to keep all results in memory, because there is a risk to cause OOM.

Reply:
Yes, we do memory control in PointCollector.